src/test/scala/is/hail/methods/SkatSuite.scala

```diff
@@ -0,0 +1,484 @@
+      val pvalR = resultsArray(i)(2).asInstanceOf[Double]
+
+
+      if (pval <= 1 && pval >= 0) {
+        assert(D_==(qstat, qstatR, tol))
+        assert(D_==(pval, pvalR, tol))
+      }
+      i += 1
+    }
+
+  }
+
+
+  //Generates random data
+  def buildWeightValues(filepath: String): Unit = {
+
+    //read in chrom:pos values
+    val fileSource = Source.fromFile(filepath)
+    val linesArray = fileSource.getLines.toArray
+    fileSource.close()
+
+    //write in randomized weights
+    val fileObject = new PrintWriter(new File(filepath))
+
+    for (i <- -1 until linesArray.size) {
+      if (i == -1) {
+        fileObject.write("Pos\tWeights\n")
+      }
+      else {
+        val pos = linesArray(i)
+        val randomWeight = scala.util.Random.nextDouble()
+        fileObject.write(s"$pos\t$randomWeight\n")
+
+      }
+    }
+    fileObject.close()
+  }
+
+  def buildCovariateMatrix(filepath: String, covariateCount: Int): Unit = {
+    val fileObject = new PrintWriter(new File(filepath))
+
+    val startIndex = 96
+    val endIndex = 116
+
+    for (i <- startIndex - 1 to endIndex) {
+
+      if (i == startIndex - 1) {
+        fileObject.write("Sample\t")
+        for (j <- 1 to covariateCount) {
+          if (j == covariateCount) {
+            fileObject.write(s"Cov$j")
+          }
+          else {
+            fileObject.write(s"Cov$j\t")
+          }
+        }
+        fileObject.write("\n")
+      }
+      else {
+        fileObject.write("HG%05d\t".format(i))
+        for (j <- 1 to covariateCount) {
+          if (j == covariateCount) {
+            fileObject.write("%d".format(scala.util.Random.nextInt(25)))
+          }
+          else {
+            fileObject.write("%d\t".format(scala.util.Random.nextInt(25)))
+          }
+        }
+        fileObject.write("\n")
+      }
+    }
+
+    fileObject.close()
+  }
+
+  //Dataset for big Test
+
+  def covariatesSkat = hc.importTable("src/test/resources/skat.cov",
+    impute = true).keyBy("Sample")
+
+  def phenotypesSkat = hc.importTable("src/test/resources/skat.pheno",
+    types = Map("Pheno" -> TFloat64), missing = "0").keyBy("Sample")
+
+  def intervalsSkat = IntervalList.read(hc, "src/test/resources/skat.interval_list")
+
+  def weightsSkat = hc.importTable("src/test/resources/skat.weights",
+    types = Map("locus" -> TLocus, "weight" -> TFloat64)).keyBy("locus")
+
+  def vdsSkat: VariantDataset = hc.importVCF("src/test/resources/sample2.vcf")
+    .annotateVariantsTable(intervalsSkat, root = "va.genes", product = true)
+    .annotateVariantsTable(weightsSkat, root = "va.weight")
+    .annotateSamplesTable(phenotypesSkat, root = "sa.pheno0")
+    .annotateSamplesTable(covariatesSkat, root = "sa.cov")
+    .annotateSamplesExpr("sa.pheno = if (sa.pheno0 == 1.0) false else if (sa.pheno0 == 2.0) true else NA: Boolean")
+
+
+  @Test def hardcallsBigTest() {
+
+    val useDosages = false
+    val covariates = new Array[String](2)
```

Comment:
This was left over from when I was randomly generating the covariates matrix during each test